pkg/networkpath/traceroute/runner.go

```diff
@@ -249,3 +329,89 @@ func getHostname(ipAddr string) string {
 	}
 	return currHost
 }
+
+func getInterface(localAddr net.IP) (net.Interface, error) {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return net.Interface{}, fmt.Errorf("failed to get interfaces: %w", err)
+	}
+
+	for _, iface := range interfaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			fmt.Printf("failed to get iface addresses: %s\n", err.Error())
+			continue
+		}
+
+		for _, addr := range addrs {
+			ip, _, err := net.ParseCIDR(addr.String())
+			if err != nil {
+				fmt.Printf("failed to parse cidr for %s: %s", addr.String(), err.Error())
+				continue
+			}
+
+			if ip.Equal(localAddr) {
+				return iface, nil
+			}
+		}
+	}
+
+	return net.Interface{}, fmt.Errorf("failed to find matching interface for %q", localAddr.String())
+}
+
+func (r *Runner) resolveSubnetID(ctx context.Context, iface net.Interface) (string, error) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Shall we have test coverage for this new feature/improvement?